test/tools/libtesteth/ImportTest.cpp

```diff
@@ -183,16 +183,16 @@ bytes ImportTest::executeTest()
 	return bytes();
 }
 
-std::pair<eth::State, ImportTest::execOutput> ImportTest::executeTransaction(eth::Network const _sealEngineNetwork, eth::EnvInfo const& _env, eth::State const& _preState, eth::Transaction const& _tr)
+std::tuple<eth::State, eth::ChangeLog, ImportTest::ExecOutput> ImportTest::executeTransaction(eth::Network const _sealEngineNetwork, eth::EnvInfo const& _env, eth::State _state, eth::Transaction const& _tr)
 {
-	eth::State initialState = _preState;
 	try
 	{
 		unique_ptr<SealEngineFace> se(ChainParams(genesisInfo(_sealEngineNetwork)).createSealEngine());
 		bool removeEmptyAccounts = m_envInfo.number() >= se->chainParams().u256Param("EIP158ForkBlock");
-		ImportTest::execOutput execOut = initialState.execute(_env, *se.get(), _tr);
-		initialState.commit(removeEmptyAccounts ? State::CommitBehaviour::RemoveEmptyAccounts : State::CommitBehaviour::KeepEmptyAccounts);
-		return std::pair<eth::State, ImportTest::execOutput>(initialState, execOut);
+		ImportTest::ExecOutput execOut = _state.execute(_env, *se.get(), _tr);
+		ChangeLog changeLog = _state.changeLog();
```

Comment:
what Permanence does ?  if tis not affecting the results perhaps we should use this flag.